### PR TITLE
#22: The solution for GenerateParentheses problem

### PR DESCRIPTION
**Pull request by ArinaSiselina**

### CODE REVIEW

File: ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿namespace LeetCodeTraining.GenerateParentheses
+{
+    // https://leetcode.com/problems/generate-parentheses/
+
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs]: Please remove extra line.

File: ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            if (current.Length == max * 2)
+            {
+                result.Add(current);
+                return;
+            }
+
+            if (open < max)
+            {
+                Backtrack(result, current + "(", open + 1, close, max);
+            }
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs:25]: Please add extra line for better code reading.

File: ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿namespace LeetCodeTraining.GenerateParentheses
+{
+    // https://leetcode.com/problems/generate-parentheses/
+
+    public class GenerateParenthesesSolution
+    {
+        public static IList<string> GenerateParentheses(int n)
+        {
+            IList<string> result = new List<string>();
+            Backtrack(result, "", 0, 0, n);
```
Comment (shomzascnsoft) [inline ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs]: Please add extra line for better code reading.
Please pass string.Empty instead ""...

File: ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs (hunk 2 above)
Comment (shomzascnsoft) [reply ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs:25]: > Please add extra line for better code reading.

Comment (ArinaSiselina) [reply ScnSoft-Education/LeetCodeTraining/GenerateParentheses-22/GenerateParenthesesSolution.cs:25]: Extra line added

**Review by shomzascnsoft — APPROVED** [review]

**Pull request merged**